neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -269,3 +275,45 @@ def create_outgoing_message_processor(self) -> MessageProcessor:
         # Eventually this might be a CompositeMessageProcessor
         message_processor = StructureMessageProcessor(structure_formats)
         return message_processor
+
+    async def close_sly_data(self):
+        """
+        Close any sly data value that can be closed.
+        """
+        if self.sly_data is None or len(self.sly_data) == 0:
+            # Nothing to close
+            return
+
+        # Try closing anything that can be close()-d in the sly_data.
+        for key, value in self.sly_data.items():
+            # Note: It's possible we might want to try closing keys as well
+            description: str = f"sly_data[{key}]"
+            await self.close_one(value, description)
+
+    async def close_one(self, value: Any, description: str):
+        """
+        Close a single object if it can be closed.
+        :param value: The object to close
+        :param description: Where the object to close came from
+        """
```

Comment:
Method to close out a single sly_data entity.
Most of the time we won't need this because most of the time sly_data is "Just Data".
But when people put objects on sly_data that have sessions and can be shared between CodedTools, then that's when we need this.